app/controllers/admin/articles_controller.rb

```diff
@@ -57,6 +66,17 @@ def toggle_status
       redirect_to admin_article_path(@article.category), notice: "Status was updated successfully."
     end
 
+    def update_groupings
+      # return if Article.where(is_part_of_main_articles: true).count > 5
+
+      @article = Article.find(params[:article_id])
+      grouping_type = params[:grouping_type]
+      @article.update(is_part_of_main_articles: true) if (grouping_type== 'grouping_main_articles')
+      @article.update(is_part_of_breakdown: true) if (grouping_type == 'grouping_breakdown')
```

Comment:
invocation of `update` two times need to be avoided, so you could build hash with attributes to you need to update  and then just have single `update` in order to avoid unnecessary queries to DB